packages/cli-repl/src/cli-repl.ts

```diff
@@ -902,6 +908,47 @@ export class CliRepl implements MongoshIOProvider {
     return this.shellHomeDirectory.roamingPath('mongosh_repl_history');
   }
 
+  /** Returns the history of commands. */
+  setupHistoryCommand(): void {
+    const getHistory = () => {
+      const replHistory: string[] = (
+        this.mongoshRepl.runtimeState().repl as any
+      ).history;
+      const formattedHistory =
+        // Remove the history call from the formatted history
+        replHistory.slice(1, replHistory.length).reverse();
+
+      return {
+        ...formattedHistory,
+        [util.inspect.custom]: () => {
+          return formatOutput(
+            {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              value: formattedHistory,
+            },
+            { colors: true, maxArrayLength: Infinity }
+          );
+        },
+      };
+    };
+
+    getHistory.isDirectShellCommand = true;
```

Comment:
Do we want this to be callable as just `history`? I don't see a major downside in making it a regular function like `history()`, and then you wouldn't need all this extra setup here either

Reply:
an argument for `history` could be that it'd make it more intuitive coming from a Unix shell so maybe it's worth it?

Reply:
For now the ability to see history and C+P historical commands sounds like a good enough value drive for customers, without having to support something complicated like `!n` in bash. For `history` vs `history()`, I don't have a strong feeling either way but `history` being intuitive from bash is a good argument for putting a little extra effort to supporting it.

Reply:
I think `history` will still be fine as far as discoverability goes as they'd see that it's a function in response and then do `history()` and use it to its fullest potential that way